crates/ast/src/visit.rs

```diff
@@ -4,139 +4,154 @@ use crate::ast::*;
 use solar_data_structures::trustme;
 use solar_interface::{Ident, Span};
 use solar_macros::declare_visitors;
+use std::ops::ControlFlow;
 
 declare_visitors! {
     /// AST traversal.
     pub trait Visit VisitMut <'ast> {
-        fn visit_source_unit(&mut self, source_unit: &#mut SourceUnit<'ast>) {
+        fn visit_source_unit(&mut self, source_unit: &#mut SourceUnit<'ast>) -> ControlFlow<()> {
             // TODO: SAFETY: Idk
             let source_unit = unsafe { trustme::decouple_lt #_mut(source_unit) };
             let SourceUnit { items } = source_unit;
             for item in items.iter #_mut() {
-                self.visit_item #_mut(item);
+                if let ControlFlow::Break(_) = self.visit_item #_mut(item) {
+                    return ControlFlow::Break(());
+                }
```

Comment:
This can be `?`

Reply:
putting a ``?" would prompt an error in the if let statement